backend/transcribee_backend/metrics.py

```diff
@@ -0,0 +1,128 @@
+import datetime
+import secrets
+from abc import abstractmethod
+from typing import List
+
+from fastapi import Depends, HTTPException, status
+from fastapi.security import HTTPBasic, HTTPBasicCredentials
+from prometheus_client import Gauge
+from sqlmodel import Session, col, func
+
+from transcribee_backend.config import settings
+from transcribee_backend.db import SessionContextManager
+from transcribee_backend.helpers.time import now_tz_aware
+from transcribee_backend.models.document import Document
+from transcribee_backend.models.task import Task, TaskAttempt, TaskState
+from transcribee_backend.models.user import User
+from transcribee_backend.models.worker import Worker
+
+
+class Metric:
+    @abstractmethod
+    def refresh(self, session: Session):
+        pass
+
+
+class TasksInState(Metric):
+    def __init__(self):
+        self.collector = Gauge("tasks", "Number of tasks", ["state"])
+
+    def refresh(self, session: Session):
+        result = session.query(Task.state, func.count()).group_by(Task.state).all()
+        counts = {x: 0 for x in TaskState}
+        for state, count in result:
+            counts[state] = count
+        for state, count in counts.items():
+            self.collector.labels(state=state.value).set(count)
+
+
+class Workers(Metric):
+    def __init__(self):
+        self.collector = Gauge("workers", "Workers", ["group"])
+
+    def refresh(self, session: Session):
+        (result,) = session.query(func.count(Worker.id)).one()
+        self.collector.labels(group="all").set(result)
+
+        now = now_tz_aware()
+        worker_timeout_ago = now - datetime.timedelta(seconds=settings.worker_timeout)
+        (result,) = (
+            session.query(func.count(Worker.id))
+            .where(
+                col(Worker.last_seen) >= worker_timeout_ago,
+            )
+            .one()
+        )
+        self.collector.labels(group="alive").set(result)
+
+
+class Users(Metric):
+    def __init__(self):
+        self.collector = Gauge("users", "Registered users")
+
+    def refresh(self, session: Session):
+        (result,) = session.query(func.count(User.id)).one()
+        self.collector.set(result)
+
+
+class Documents(Metric):
+    def __init__(self):
+        self.collector = Gauge("documents", "Documents")
+
+    def refresh(self, session: Session):
+        (result,) = session.query(func.count(Document.id)).one()
+        self.collector.set(result)
+
+
+class Queue(Metric):
+    def __init__(self):
+        self.collector = Gauge("queue", "Queue length in seconds")
```

Comment:
Same here. I would like to have every task chain only counted once

Reply:
Or maybe we introduce some unit here that is not seconds (maybe something like estimated compute seconds on some specific platform? the mac mini?)

Reply:
Yes and no. Doing a better calculation that takes into account the real time factor of different machines etc would be a nice addition, but I'd rather do that in a seperate PR, as this has some more open questions to me (and probably needs some database changes as well)

Reply:
Then for the time being only count transcription tasks?